src/openrct2/CmdlineSprite.cpp

```diff
@@ -90,17 +90,16 @@ static bool sprite_file_open(const utf8* path)
 
     if (spriteFileHeader.num_entries > 0)
     {
-        int32_t openEntryTableSize = spriteFileHeader.num_entries * sizeof(rct_g1_element_32bit);
-        rct_g1_element_32bit* openElements = static_cast<rct_g1_element_32bit*>(malloc(openEntryTableSize));
+        std::unique_ptr<rct_g1_element_32bit[]> openElements = std::make_unique<rct_g1_element_32bit[]>(
+            spriteFileHeader.num_entries);
         if (openElements == nullptr)
         {
             fclose(file);
             return false;
         }
 
-        if (fread(openElements, openEntryTableSize, 1, file) != 1)
+        if (fread(openElements.get(), spriteFileHeader.num_entries, 1, file) != 1)
```

Comment:
This is now incorrect, fread expects the total amount of bytes as the parameter, this passes only the amount of entries.

Reply:
Sorry about that! This should fix it